simtools/simtel/simtel_runner_array.py

```diff
@@ -121,92 +121,102 @@ def _load_simtel_data_directories(self):
         self._simtel_log_dir = simtel_base_dir.joinpath("log")
         self._simtel_log_dir.mkdir(parents=True, exist_ok=True)
 
-    def get_log_file(self, run_number):
-        """Get full path of the simtel log file for a given run."""
-        file_name = names.simtel_log_file_name(
-            run=run_number,
-            primary=self.config.primary,
-            array_name=self.array_model.layout_name,
-            site=self.array_model.site,
-            zenith=self.config.zenith_angle,
-            azimuth=self.config.azimuth_angle,
-            label=self.label,
-        )
-        return self._simtel_log_dir.joinpath(file_name)
+    def get_info_for_file_name(self, run_number):
+        """
+        Get a dirctionary with the info necessary for building the sim_telarray file names.
 
-    def get_sub_log_file(self, run_number, mode="out"):
+        Returns
+        -------
+        dict
+            Dictionary with the keys necessary for building the sim_telarray file names.
         """
-        Get the full path of the submission log file.
+        return {
+            "run": run_number,
+            "primary": self.config.primary,
+            "array_name": self.array_model.layout_name,
+            "site": self.array_model.site,
+            "zenith": self.config.zenith_angle,
+            "azimuth": self.config.azimuth_angle,
+            "label": self.label,
+        }
+
+    def get_file_name(self, file_type, **kwargs):
+        """
+        Get a sim_telarray style file name for various file types.
 
         Parameters
         ----------
-        run_number: int
-            Run number.
-        mode: str
-            out or err
+        file_type: str
+            The type of file it is (determines the file suffix).
+            Choices are log, histogram, output or sub_log.
+        kwargs: dict
+            The dictionary must include the following parameters (unless listed as optional):
+                run: int
+                    Run number.
+                primary: str
+                    Primary particle (e.g gamma, proton etc).
+                zenith: float
+                    Zenith angle (deg).
+                azimuth: float
+                    Azimuth angle (deg).
+                site: str
+                    Paranal or LaPalma.
```

Comment:
Wondering if we need to say here that it has to be Paranal or La Palma. The code would work with any site, or?

Reply:
Yes, this specific function would work with any site so I could remove this. In general we of course force it to be Paranal or La Palma so it would fail elsewhere I assume. Either way, removed it from here.